lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -2940,15 +2992,16 @@ struct MMA16816ConversionHelper {
 
     assert(!elems.empty());
 
-    Type fp16Ty = aTensorTy.getElementType();
+    Type fp16Ty = type::f16Ty(ctx);
```

Comment:
1, nit: it's better to always have type from TypeConverter/ConversionPatternRewriter, than directly created from MLIRContext.
2, why do we need to explicitly fetch f16Ty here instead of deriving it from the operand? which means that we have to do something in addition for other dtypes like int8.

Reply:
The `ldmatrix` in mma<version=2> can only work on b16, that's why here can only be a f16.

Ref https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#warp-level-matrix-instructions-ldmatrix